### PR TITLE
Expose time utils.

### PR DESCRIPTION
**Pull request by Shylock-Hg**
Expose the time utils which used in cpp-client.

### CODE REVIEW

File: src/common/base/Status.h [hunk 1]
```diff
@@ -234,15 +237,15 @@ inline std::ostream& operator<<(std::ostream &os, const Status &status) {
 #define NG_RETURN_IF_ERROR(s)                                                                      \
     do {                                                                                           \
         const auto &__s = (s);                                                                     \
-        if (UNLIKELY(!__s.ok())) {                                                                 \
+        if (!__s.ok()) {                                                                           \
```
Comment (Aiee) [inline src/common/base/Status.h:240]: Why delete this?

File: src/common/time/TimeUtils.cpp [hunk 2]
```diff
@@ -272,5 +243,49 @@ StatusOr<Value> TimeUtils::toTimestamp(const Value &val) {
...
+/*static*/ DateTime TimeUtils::dateTimeToUTC(const DateTime &dateTime) {
+    return dateTimeShift(dateTime, -Timezone::getGlobalTimezone().utcOffsetSecs());
+}
+
+/*static*/ DateTime TimeUtils::utcToDateTime(const DateTime &dateTime) {
+    return dateTimeShift(dateTime, Timezone::getGlobalTimezone().utcOffsetSecs());
+}
+
+/*static*/ StatusOr<DateTime> TimeUtils::localDateTime() {
+    DateTime dt;
```
Comment (laura-ding) [inline src/common/time/TimeUtils.cpp:256]: unused

File: src/common/time/TimeUtils.cpp [hunk 3]
```diff
@@ -272,5 +243,49 @@ StatusOr<Value> TimeUtils::toTimestamp(const Value &val) {
...
+    time_t unixTime = std::time(NULL);
+    if (unixTime == -1) {
+        return Status::Error("Get unix time failed: %s.", std::strerror(errno));
+    }
+    return unixSecondsToDateTime(unixTime - Timezone::getGlobalTimezone().utcOffsetSecs());
+}
+
+/*static*/ StatusOr<Date> TimeUtils::localDate() {
+    Date d;
```
Comment (laura-ding) [inline src/common/time/TimeUtils.cpp:265]: ditto

File: src/common/time/TimeUtils.cpp [hunk 4]
```diff
@@ -272,5 +243,49 @@ StatusOr<Value> TimeUtils::toTimestamp(const Value &val) {
...
+/*static*/ Time TimeUtils::timeToUTC(const Time &time) {
+    return timeShift(time, -Timezone::getGlobalTimezone().utcOffsetSecs());
+}
+
+/*static*/ Time TimeUtils::utcToTime(const Time &time) {
+    return timeShift(time, Timezone::getGlobalTimezone().utcOffsetSecs());
+}
+
+/*static*/ StatusOr<Time> TimeUtils::localTime() {
+    Time dt;
```
Comment (laura-ding) [inline src/common/time/TimeUtils.cpp:282]: ditto

File: src/common/time/TimeUtils.cpp [hunk 5]
```diff
@@ -272,5 +243,49 @@ StatusOr<Value> TimeUtils::toTimestamp(const Value &val) {
...
+
+/*static*/ DateTime TimeUtils::dateTimeToUTC(const DateTime &dateTime) {
+    return dateTimeShift(dateTime, -Timezone::getGlobalTimezone().utcOffsetSecs());
+}
+
+/*static*/ DateTime TimeUtils::utcToDateTime(const DateTime &dateTime) {
+    return dateTimeShift(dateTime, Timezone::getGlobalTimezone().utcOffsetSecs());
+}
+
+/*static*/ StatusOr<DateTime> TimeUtils::localDateTime() {
```
Comment (laura-ding) [inline src/common/time/TimeUtils.cpp:255]: `LocalDateTime ()` does not need to be exposed to the user, so `StatusOr` and `Status` does not need to be exposed to the cpp client

File: src/common/base/Status.h (hunk 1 above)
Comment (Shylock-Hg) [reply src/common/base/Status.h:240]: Used in cpp client.

File: src/common/time/TimeUtils.cpp (hunk 2 above)
Comment (Shylock-Hg) [reply src/common/time/TimeUtils.cpp]: Fixed.

File: src/common/time/TimeUtils.cpp (hunk 5 above)
Comment (Shylock-Hg) [reply src/common/time/TimeUtils.cpp:255]: Many method return status.

File: src/common/base/StatusOr.h [hunk 6]
```diff
@@ -230,12 +233,12 @@ class StatusOr final {
     // Return the associated `Status' if and only if it has one,
     //
     Status status() const & {
-        CHECK(hasStatus());
+        assert(hasStatus());
```
Comment (laura-ding) [inline src/common/base/StatusOr.h:236]: The Status no need to expose to the client, you can revert it.

Comment (Shylock-Hg) [reply src/common/base/StatusOr.h]: Fixed.

File: src/common/time/TimeConversion.cpp [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+#include "common/time/TimeConversion.h"
+
+namespace nebula {
+namespace time {
+
+const DateTime TimeConversion::kEpoch(1970, 1, 1, 0, 0, 0, 0);
+
+/*static*/ int64_t TimeConversion::dateTimeDiffSeconds(const DateTime &dateTime0,
+                                                  const DateTime &dateTime1) {
```
Comment (laura-ding) [inline src/common/time/TimeConversion.cpp:15]: alignment

**Review by laura-ding — APPROVED** [review]

File: src/common/time/TimeConversion.cpp (hunk 7 above)
Comment (Shylock-Hg) [reply src/common/time/TimeConversion.cpp]: Fixed.

**Review by laura-ding — APPROVED** [review]

**Review by Aiee — APPROVED** [review]

**Pull request merged**